### PR TITLE
timescaledb: Move unfree library to separate package to fix the license

### PR DESCRIPTION
**Pull request by chkno**
###### Motivation for this change
Fix the timescaledb license.
Make the Free/Libre/Open Source timescaledb build available.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
    43361488 → 43005136
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

timescaledb maintainers: @volth @marsam

### CODE REVIEW

File: pkgs/servers/sql/postgresql/ext/timescaledb.nix [hunk 1]
```diff
@@ -1,4 +1,5 @@
-{ stdenv, fetchFromGitHub, cmake, postgresql, openssl }:
+{ stdenv, fetchFromGitHub, cmake, postgresql, openssl
+, build-unfree-tsl ? false }:
```
Comment (nora-puchreiner) [inline pkgs/servers/sql/postgresql/ext/timescaledb.nix:2]:
```suggestion
, build-unfree-tsl }:
```

File: pkgs/servers/sql/postgresql/packages.nix [hunk 2]
```diff
@@ -53,6 +53,9 @@ self: super: {
 
     timescaledb = super.callPackage ./ext/timescaledb.nix { };
```
Comment (nora-puchreiner) [inline pkgs/servers/sql/postgresql/packages.nix:54]:
```suggestion
    timescaledb = super.callPackage ./ext/timescaledb.nix {  build-unfree-tsl = false; };
```

**Review by nora-puchreiner — CHANGES_REQUESTED** [review]
Please, make the choice explicit.
Defaulting to any one of the choices is counter-intuitive
One might expect unfree version by default having `nixpkgs.config.allowUnfree = true`

**Comment by chkno** [comment]
> Please, make the choice explicit.
> Defaulting to any one of the choices is counter-intuitive
> One might expect unfree version by default having `nixpkgs.config.allowUnfree = true`

Done.  Thanks!

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 103507` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package blacklisted:</summary>
  <ul>
    <li>tests.nixos-functions.nixos-test</li>
  </ul>
</details>
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>postgresqlPackages.timescaledb (postgresql11Packages.timescaledb)</li>
    <li>postgresqlPackages.timescaledb-tsl (postgresql11Packages.timescaledb-tsl)</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 103507` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>postgresqlPackages.timescaledb (postgresql11Packages.timescaledb)</li>
    <li>postgresqlPackages.timescaledb-tsl (postgresql11Packages.timescaledb-tsl)</li>
  </ul>
</details>

**Pull request closed without merging**